### PR TITLE
Update SPneg.mdx

### PR DESCRIPTION
**Pull request by leoleo28**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [v] I have tested my code.
- [v] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [v] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/6_Advanced/SPneg.mdx [hunk 1]
```diff
@@ -215,6 +215,89 @@ int main() {
...
+			for (int x : cycle) { io.print(x + " "); }
+			io.println();
+		}
+		io.close();
+	}
+
+	// CodeSnip{Kattio}
+}
+
+class Edge {
```
Comment (SansPapyrus683) [inline content/6_Advanced/SPneg.mdx:285]: 1. code snip this
2. i'd put this in the main class tbh (it would have to be static)

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/SPneg.mdx [hunk 2]
```diff
@@ -215,6 +215,89 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Main {
```
[the remark this answers is not in the document]
Comment (leoleo28) [reply content/6_Advanced/SPneg.mdx]: Fix it

File: content/6_Advanced/SPneg.mdx (hunk 1 above)
Comment (leoleo28) [reply content/6_Advanced/SPneg.mdx]: Fix it.

File: content/6_Advanced/SPneg.mdx [hunk 3]
```diff
@@ -221,14 +221,27 @@ int main() {
...
-public class Main {
+public class CyclesFinding {
 	static final int N = 2501;
 	static int n, m;
 	static int[] parent = new int[N];
 	static long[] dist = new long[N];
 	static List<Edge> graph = new ArrayList<>();
 	static Kattio io = new Kattio();
 
+	// CodeSnip{Edge}
```
Comment (SansPapyrus683) [inline content/6_Advanced/SPneg.mdx:232]: you need a `// EndCodeSnip` too

Comment (leoleo28) [reply content/6_Advanced/SPneg.mdx:232]: Ok, I add it

File: content/6_Advanced/SPneg.mdx [hunk 4]
```diff
@@ -215,6 +215,91 @@ int main() {
...
+		for (int i = 1; i <= m; i++) {
+			int a = io.nextInt();
+			int b = io.nextInt();
+			long c = (long)io.nextInt();
+			graph.add(new Edge(a, b, c));
+		}
+		bellmanFord(1);
+	}
+
+	private static void bellmanFord(int source) {
```
Comment (SansPapyrus683) [inline content/6_Advanced/SPneg.mdx:257]: the void here is kinda sus
imo it should return the cycle if there is one, and null if there isn't
`main` should then check for it and output accordingly
makes the flow of the program much better

**Review by envyaims — APPROVED** [review]
looks fire

File: content/6_Advanced/SPneg.mdx [hunk 5]
```diff
@@ -214,7 +214,89 @@ int main() {
...
 
 </CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class CyclesFinding {
+	static final int N = 2501;
```
Comment (ryanchou-dev) [inline content/6_Advanced/SPneg.mdx:224]: make $n$ 2500 and add $1$ when you create your arrays

**Review by ryanchou-dev — APPROVED** [review]
cool!

File: content/6_Advanced/SPneg.mdx (hunk 5 above)
Comment (leoleo28) [reply content/6_Advanced/SPneg.mdx]: Fix it

File: content/6_Advanced/SPneg.mdx [hunk 6]
```diff
@@ -214,7 +214,94 @@ int main() {
...
+import java.io.*;
+import java.util.*;
+
+public class CyclesFinding {
+	static final int N = 2500;
+	static int n, m;
+	static int[] parent = new int[N + 1];
+	static long[] dist = new long[N + 1];
+	static List<Edge> graph = new ArrayList<>();
+	static Kattio io = new Kattio();
```
Comment (SansPapyrus683) [inline content/6_Advanced/SPneg.mdx:229]: i feel like a lot of these don't actually need to be static

Comment (leoleo28) [reply content/6_Advanced/SPneg.mdx:229]: But without static keyword, those variables cann't be referenced from the static method (main).

Comment (SansPapyrus683) [reply content/6_Advanced/SPneg.mdx:229]: make them local to the function (except the ones the dfs uses obviously)

Comment (leoleo28) [reply content/6_Advanced/SPneg.mdx]: fix it

**Review by SansPapyrus683 — APPROVED** [review]
alright! show final screenshot of AC (with the code included in the screenshot!!!!) and i'll merge

**Comment by leoleo28** [comment]
> alright! show final screenshot of AC (with the code included in the screenshot!!!!) and i'll merge

Here is the screenshot of updated code.  

截圖 2024-01-07 上午10 50 12
截圖 2024-01-07 上午10 50 18

**Pull request merged**